File: scharp/day/five/SeleniumTestProject/pages/WebFormPage.cs
using OpenQA.Selenium;

namespace SeleniumTestProject.pages
{
    public class WebFormPage : BasePage 
    {
        private readonly String _url = "https://www.selenium.dev/selenium/web/web-form.html";
        private readonly By _textInput = By.Name("my-text");
        private readonly By _submitButton = By.CssSelector("button[type='submit']");
        public WebFormPage(IWebDriver driver, int waitInSeconds) : base(driver, waitInSeconds) { }

        public WebFormPage Open()
        {
            driver.Navigate().GoToUrl(_url);
            wait.Until(driver =>
            {
                IWebElement textInput = driver.FindElement(_textInput);
                return textInput.Displayed ? textInput : null;
            });
            return this;
        }

        public WebFormPage TypeText(String text) {
            IWebElement textInput = driver.FindElement(_textInput);
            textInput.Clear();
            textInput.SendKeys(text);
            return this;
        }

        public SubmittedPage Submit()
        {
            IWebElement submit = wait.Until(driver =>
            {
                IWebElement submit = driver.FindElement(_submitButton);
                return submit.Displayed ? submit : null;
            });
            submit.Click();
            return new SubmittedPage(driver, waitDurationSeconds);
        }

    }
}

File: java/java-practise/src/main/java/day/five/selenium-page-object/src/test/java/pages/WebFormPage.java
package pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.ui.ExpectedConditions;

public class WebFormPage extends BasePage {

    private static final String URL = "https://www.selenium.dev/selenium/web/web-form.html";
    private final By textInputLocator = By.name("my-text");
    private final By submitButton = By.cssSelector("button[type='submit']");

    public WebFormPage(WebDriver driver, int waitInSeconds){
        super(driver, waitInSeconds);
    }

    public WebFormPage open(){
        driver.get(URL);
        wait.until(ExpectedConditions.visibilityOfElementLocated(textInputLocator));
        return this;
    }

    public WebFormPage typeText(String text){
        WebElement input = driver.findElement(textInputLocator);
        input.clear();
        input.sendKeys(text);
        return this;
    }

    public SubmittedPage submit(){
        WebElement button = wait.until(ExpectedConditions.visibilityOfElementLocated(submitButton));
        button.click();
        return new SubmittedPage(driver, waitDuration);
    }

}
